libmscore/glissando.cpp

```diff
@@ -370,6 +380,21 @@ void Glissando::write(Xml& xml) const
       if (_showText && !_text.isEmpty())
             xml.tag("text", _text);
       xml.tag("subtype", int(_glissandoType));
+      writeProperty(xml, P_ID::PLAY_GLISSANDO);
+      switch ( glissandoStyle()) {
+          case MScore::GlissandoStyle::BLACK_KEYS:
```

Comment:
I.e., where as the write function follows the pattern

writeProperty*(xml, P_ID::something)

I’d expect the read pattern to be something like.

setSomething( readProperty( xml, P_ID::something));


> On 14 May 2015, at 11:22, Jim Newton <jimka.issy@gmail.com> wrote:
> 
> HI Nicolas, I found the example of how Direction is handled form write-xml.  But I didn’t find where it is handled for read-xml.
> Can you point me to that example please?
> 
> Thanks.
> Jim
> 
> 
>> On 13 May 2015, at 11:55, Nicolas Froment <notifications@github.com <mailto:notifications@github.com>> wrote:
>> 
>> In libmscore/glissando.cpp <https://github.com/musescore/MuseScore/pull/1946#discussion_r30216878>:
>> 
>> > @@ -370,6 +380,21 @@ void Glissando::write(Xml& xml) const
>> >        if (_showText && !_text.isEmpty())
>> >              xml.tag("text", _text);
>> >        xml.tag("subtype", int(_glissandoType));
>> > +      writeProperty(xml, P_ID::PLAY_GLISSANDO);
>> > +      switch ( glissandoStyle()) {
>> > +          case MScore::GlissandoStyle::BLACK_KEYS:
>> Well in fact is not a very good idea. A good solution for storing strings instead of enum casted to int would be to create a Property Type like it's done for Direction. Check https://github.com/musescore/MuseScore/blob/master/libmscore/xml.cpp#L394 <https://github.com/musescore/MuseScore/blob/master/libmscore/xml.cpp#L394> to see how the property types are handled
>> 
>> —
>> Reply to this email directly or view it on GitHub <https://github.com/musescore/MuseScore/pull/1946/files#r30216878>.
>> 
>